### PR TITLE
Move `cache_tree_children` to `TreeQuerySet.get_cached_trees`.

### PR DESCRIPTION
**Pull request by danielnaab**
@craigds, for #413 it seemed most appropriate to put this on the `TreeQuerySet` rather than on the `Manager`, as it allows one to get an entire tree with one query (you don't need to query the root nodes first, then children). The existing code works in this context with very little change, and I left the current `cache_tree_children` tag in place with a `DeprecationWarning`.

Let me know how this looks, or if needs additional changes.

### CODE REVIEW

File: tests/myapp/tests.py [hunk 1]
```diff
@@ -1806,3 +1806,51 @@ def test_insert_ordered_BFS_backwards_nonroot_nodes(self):
...
+
+        # Getting the ancestors of each node should not result in db hits.
+        with self.assertNumQueries(0):
+            self.assertEqual(len(root_nodes), 1)
+            category = root_nodes[0]
+            self.assertIsInstance(category, Category)
+            self.assertEqual(len(category.get_children()), 3)
+            for child in category.get_children():
+                self.assertIsInstance(child, Category)
+                for child in category.get_children():
```
Comment (craigds) [inline tests/myapp/tests.py]: was this meant to iterate over the children of `child` instead of `category`?

File: tests/myapp/tests.py [hunk 2]
```diff
@@ -1806,3 +1806,51 @@ def test_insert_ordered_BFS_backwards_nonroot_nodes(self):
...
+        """
+        Test a query with one root node.
+        """
+        with self.assertNumQueries(1):
+            root_nodes = Category.objects.all().get_cached_trees()
+
+        # `get_cached_trees` should only return the root nodes
+        self.assertEqual(len(root_nodes), 1)
+
+        # Getting the ancestors of each node should not result in db hits.
```
Comment (craigds) [inline tests/myapp/tests.py]: Looks like this only tests `get_children()` but the comment says ancestors

File: tests/myapp/tests.py [hunk 3]
```diff
@@ -1806,3 +1806,51 @@ def test_insert_ordered_BFS_backwards_nonroot_nodes(self):
...
+            self.assertEqual(len(root_nodes), 1)
+            category = root_nodes[0]
+            self.assertIsInstance(category, Category)
+            self.assertEqual(len(category.get_children()), 3)
+            for child in category.get_children():
+                self.assertIsInstance(child, Category)
+                for child in category.get_children():
+                    self.assertIsInstance(child, Category)
+
+    def test_genre_iter(self):
```
Comment (craigds) [inline tests/myapp/tests.py]: Does this test a different edge case than the other test?

If not, maybe just have one of them :)

**Comment by craigds** [comment]
Thanks, awesome work! A couple of tweaks to the tests (as noted) and then I think I can merge this :smile:

**Comment by craigds** [comment]
Looks great. One more thing: should probably add this method to the docs :)

**Comment by danielnaab** [comment]
@craigds The latest commit has some docs. I also lightly filled in some bits that were missing elsewhere.
